examples/ipfs-kad/src/main.rs

```diff
@@ -110,3 +126,62 @@ async fn main() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+async fn insert_pk_record<TStore: kad::store::RecordStore + Send + 'static>(
+    swarm: &mut Swarm<kad::Behaviour<TStore>>,
```

Comment:
I am a bit worried that we are teaching the wrong concepts here. Instead of having two functions that poll the `Swarm`, it might be better to introduce a minimal version of something like https://github.com/libp2p/rust-libp2p/blob/master/examples/file-sharing/src/network.rs.

Reply:
How about we simplify it to a single loop at least?

We can instruct the `Swarm` what to do before we start the loop and then handle both events in there. That should be reasonably concise :)

Reply:
Done